### PR TITLE
[github] - update package-spec to 2.9.0

### PR DESCRIPTION
**Pull request by taylor-swanson**
## What does this PR do?

- Updates package-spec to 2.9.0
- Change `github.issues.assignees` type to flattened
- Change `github.issues.labels` type to group

```
[git-generate]
go run github.com/andrewkroh/go-examples/ecs-update@latest -ecs-version=8.8.0 -ecs-git-ref=v8.8.0 -format-version=2.9.0 packages/github
```

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates elastic/security-team#5870

### CODE REVIEW

File: packages/github/data_stream/issues/fields/fields.yml [hunk 1]
```diff
@@ -81,10 +81,10 @@
             - name: site_admin
               type: boolean
         - name: assignees
-          type: array
+          type: flattened
```
Comment (taylor-swanson) [inline packages/github/data_stream/issues/fields/fields.yml:84]: I suppose this could also be `nested` or `object`. Any thoughts?

Comment (efd6) [reply packages/github/data_stream/issues/fields/fields.yml:84]: Have there been requests for searching on this?

File: packages/github/data_stream/issues/fields/fields.yml [hunk 2]
```diff
@@ -81,10 +81,10 @@
...
           description: "Information of users who were assigned the issue\n  \n"
```
Comment (efd6) [inline packages/github/data_stream/issues/fields/fields.yml:85]: Remove the `\n`s while you are here?

File: packages/github/changelog.yml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.17.0"
+  changes:
+    - description: Update package-spec to 2.9.0.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/1
```
Comment (efd6) [inline packages/github/changelog.yml:6]:
```suggestion
      link: https://github.com/elastic/integrations/pull/6986
```

File: packages/github/data_stream/issues/fields/fields.yml (hunk 1 above)
Comment (taylor-swanson) [reply packages/github/data_stream/issues/fields/fields.yml:84]: Not that I'm aware of. The previous type was `array` and I'm not entirely sure what elasticsearch does with that. Package spec flagged it as an invalid type, hence why it was changed here.

Comment (efd6) [reply packages/github/data_stream/issues/fields/fields.yml:84]: Yeah. I generally go to flattened first and then if we need to search on it, think about nested.

**Pull request merged**